### PR TITLE
Add- Navigate Navbar, RoutePath, Route, HomePage

### PR DESCRIPTION
**Pull request by KwinyarutP**
Navigate link in Navbar to other page and fetch api search suggestion.

### CODE REVIEW

File: client/src/components/Navbaruser.jsx [hunk 1]
```diff
@@ -8,23 +10,27 @@ const Navbaruser = () => {
...
               {/* Navigation Links */}
-              <div className="hidden md:ml-6 md:flex md:space-x-8">
-                <a href="#" className="text-gray-900 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">คำศัพท์</a>
-                <a href="#" className="text-gray-900 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">สนับสนุน</a>
-                <a href="#" className="text-gray-900 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">เกี่ยวกับเรา</a>
+              <div className="md:ml-6 md:flex md:space-x-8">
+                <Link to="/vocabulary" className="text-gray-900 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">คำศัพท์</Link>
+                <Link to="/support" className="text-gray-900 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">สนับสนุน</Link>
+                <Link to="/aboutus" className="text-gray-900 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">เกี่ยวกับเรา</Link>
               </div>
```
Comment (Mark169Z) [inline client/src/components/Navbaruser.jsx:24]: Move some classes to parent div

File: client/src/general/Router.jsx [hunk 2]
```diff
@@ -13,6 +13,12 @@ import { Layout } from "./Layout.jsx";
 import Navbaruser from "../components/Navbaruser.jsx";
 import { Vocabulary } from "../pages/vocabulary/VocabularyPage.jsx";
 import { AdminTicketPage } from "../pages/admin/AdminTicketPage.jsx"
+// Will create all page below after get UI
+// import { Support } from "../pages/support/SupportPage.jsx";
+// import { AboutUs } from "../pages/aboutus/AboutUsPage.jsx";
+// import { Register } from "../pages/register/RegisterPage.jsx";
+// import { Login } from "../pages/login/LoginPage.jsx";
+
```
Comment (Mark169Z) [inline client/src/general/Router.jsx:21]: Okay

File: client/src/components/Searchbar.jsx [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
 function SearchBar({
   selectedSearch,
   setSelectedSearch,
   setQuery,
+  searchResults,
 }) {
 
   return (
-    <Combobox value={selectedSearch} onChange={setSelectedSearch}>
+    <Combobox value={selectedSearch} onChange={setSelectedSearch}>
```
Comment (Mark169Z) [inline client/src/components/Searchbar.jsx:11]: I think this one can be simplified. up to you

**Review by Mark169Z — COMMENTED** [review]
In some parts of the code, you can avoid use redundant class by putting the class in parent div, and some part can be simplified. but overall code is good. Great!

File: client/src/pages/home/HomePage.jsx [hunk 4]
```diff
@@ -1,43 +1,64 @@
...
+          <SearchBar
             selectedSearch={selectedSearch}
             setSelectedSearch={setSelectedSearch}
             setQuery={setQuery}
-            filteredSearchResults={filteredSearchResults}
-            />
-            <CustomBtn label="ค้นหา"/>
-          </div>
-          </div>
+            // filteredSearchResults={filteredSearchResults}
```
Comment (NANAYEMYINTKYI) [inline client/src/pages/home/HomePage.jsx:56]: Do you still used filtersearchresult or not ??

File: client/src/components/Searchbar.jsx [hunk 5]
```diff
@@ -1,21 +1,28 @@
...
-    <Combobox value={selectedSearch} onChange={setSelectedSearch}>
+    <Combobox value={selectedSearch} onChange={setSelectedSearch}> 
       <div className="relative w-full mx-auto">
         <Combobox.Input
           className="w-full py-2 pl-3 pr-10 text-sm leading-5 text-gray-900 bg-white border-2 border-gray-200 rounded-full focus:outline-none focus:ring-1 focus:ring-blue-500 focus:border-blue-500"
           onChange={(event) => setQuery(event.target.value)}
           displayValue={(item) => (item ? item.name : '')}
           placeholder="ค้นหา..."
         />
+        <Combobox.Options>
```
Comment (NANAYEMYINTKYI) [inline client/src/components/Searchbar.jsx:19]: Suggest to change the variable name person to relate name to the function

**Review by NANAYEMYINTKYI — APPROVED** [review]

File: client/src/components/Navbaruser.jsx (hunk 1 above)
Comment (KwinyarutP) [reply client/src/components/Navbaruser.jsx:24]: done

File: client/src/pages/home/HomePage.jsx (hunk 4 above)
Comment (KwinyarutP) [reply client/src/pages/home/HomePage.jsx:56]: delete

File: client/src/components/Searchbar.jsx (hunk 3 above)
Comment (KwinyarutP) [reply client/src/components/Searchbar.jsx:11]: done

**Review by Chatchaponz — CHANGES_REQUESTED** [review]

**Review by Chatchaponz — APPROVED** [review]

**Pull request merged**